src/Extensions/Abstractions/Activities/ActivityExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Omex.Extensions.Abstractions.Activities
+{
+	/// <summary>
+	/// Extensions for Activity
+	/// </summary>
+	public static class ActivityExtensions
+	{
+		/// <summary>
+		/// Get user hash from activity
+		/// </summary>
+		public static string GetUserHash(this Activity activity) =>
+			activity.GetBaggageItem(UserHashKey) ?? string.Empty;
+
+
+		/// <summary>
+		/// Returns true if activity is transaction
+		/// </summary>
+		public static bool IsTransaction(this Activity activity) =>
+			string.Equals(
+				activity.GetBaggageItem(TransactionMarkerKey),
+				TransactionMarkerValue,
+				StringComparison.OrdinalIgnoreCase);
+
+
+		/// <summary>
+		/// Set user hash for the activity
+		/// </summary>
+		/// <remarks>This property would be transfered to child activity and via web requests</remarks>
+		public static Activity SetUserHash(this Activity activity, string userHash) =>
+			activity.AddBaggage(UserHashKey, userHash);
+
+
+		/// <summary>
+		/// Mark activity as transaction
+		/// </summary>
+		/// <remarks>This property would be transfered to child activity and via web requests</remarks>
+		public static Activity MarkAsTransaction(this Activity activity) =>
+			activity.AddBaggage(TransactionMarkerKey, TransactionMarkerValue);
+
+
+		/// <summary>
+		/// Get correlation guid that is used by old Omex services
+		/// </summary>
+		[Obsolete(CorrelationIdObsoleteMessage, false)]
+		public static Guid? GetObsoleteCorrelationId(this Activity activity) =>
+			Guid.TryParse(activity.GetBaggageItem(ObsoleteCorrelationId), out Guid correlation)
```

Comment:
>correlation [](start = 74, length = 11)

I assume this API returns `Guid.Empty` in the case of failures. Are we sure this isn't sufficient for the obsolete API? Do we definitely need to switch this to `null` in this case?

Reply:
I think `null` in this case better indicates that value not set, I'm not sure that default values are not used in other context, but you might be right

Reply:
That makes sense. I only called it out because often people use null when `Guid.Empty` or similar would be sufficient for their needs

---
In reply to: [394707906](https://github.com/microsoft/Omex/pull/169#discussion_r394707906) [](ancestors = 394707906)